sbi/inference/snpe/snpe_b.py

```diff
@@ -97,11 +97,7 @@ def _get_log_prob_proposal_posterior(
         self._assert_all_finite(log_prob_proposal, "proposal posterior eval")
 
         # Compute log prob with importance weights.
-        log_prob = (
-            self.calibration_kernel(x)
-            * torch.exp(log_prob_prior - log_prob_proposal)
-            * log_prob_posterior
-        )
+        log_prob = torch.exp(log_prob_prior - log_prob_proposal) * log_prob_posterior
```

Comment:
Is the subtraction in the exponent happening for numerical stability (vs. prob_prior/prob_proposal)? I would like a little comment about that. This would be my suggestion:
> `# Lueckmann et al. 2017, Eq. 1. Importance factor calculated in exp space for numerical stability.`

Reply:
So you would suggest:
`torch.exp(log_prob_prior) / torch.exp(log_prob_proposal)`?

Importance weights are in non-log-space, so there is no way around using the `exp` somewhere.

Reply:
No, no, it's better to have subtractions than divisions (e.g. when proposal puts low prob). It's all good, I forgot we get probs always in log space, and then subtracting inside the exp is the natural thing to do.